stack/core/src/main/java/org/apache/usergrid/corepersistence/CpEntityManagerFactory.java

```diff
@@ -317,12 +317,11 @@ public void deleteApplication(UUID applicationId) throws Exception {
         Query q = Query.fromQL(String.format("select * where applicationUuid = '%s'", applicationId.toString()));
         Results results = em.searchCollection( em.getApplicationRef(), "appinfos", q);
         Entity appToDelete = results.getEntity();
-        em.delete( appToDelete );
-
-        // create new Entity in deleted_appinfos collection, with same UUID and properties as deleted appinfo
-        em.create( "deleted_appinfo", appToDelete.getProperties() );
-
-        em.refreshIndex();
+        if(appToDelete != null) {
+            em.delete(appToDelete);
```

Comment:
We should reverse this.  If the node dies after this call, we won't have marked it.  Better to have an erroneous mark than to miss it completely.